src/Severity.php

```diff
@@ -0,0 +1,157 @@
+     */
+    public const SEVERITY_DEBUG = 'debug';
+
+    /**
+     * This constant represents the "info" severity level.
+     */
+    public const SEVERITY_INFO = 'info';
+
+    /**
+     * This constant represents the "warning" severity level.
+     */
+    public const SEVERITY_WARNING = 'warning';
+
+    /**
+     * This constant represents the "error" severity level.
+     */
+    public const SEVERITY_ERROR = 'error';
+
+    /**
+     * This constant represents the "fatal" severity level.
+     */
+    public const SEVERITY_FATAL = 'fatal';
+
+    /**
+     * This constant contains the list of allowed enum values.
+     */
+    public const ALLOWED_SEVERITIES = [
+        self::SEVERITY_DEBUG,
+        self::SEVERITY_INFO,
+        self::SEVERITY_WARNING,
+        self::SEVERITY_ERROR,
+        self::SEVERITY_FATAL,
+    ];
+
+    /**
+     * @var string The value of this enum instance
+     */
+    private $value;
+
+    /**
+     * Constructor.
+     *
+     * @param string $value The value this instance represents
+     */
+    private function __construct(string $value = self::SEVERITY_INFO)
+    {
+        if (!\in_array($value, self::ALLOWED_SEVERITIES, true)) {
+            throw new \InvalidArgumentException(sprintf('The "%s" is not a valid enum value.', $value));
+        }
+
+        $this->value = $value;
+    }
+
+    /**
+     * Creates a new instance of this class with the given severity value.
+     *
+     * @param string $severity The severity
+     *
+     * @return self
+     */
+    public static function fromString(string $severity): self
+    {
+        return new self($severity);
+    }
+
+    /**
+     * Creates a new instance of this enum for the "debug" value.
+     *
+     * @return self
+     */
+    public static function debug(): self
+    {
+        return new self(self::SEVERITY_DEBUG);
+    }
+
+    /**
+     * Creates a new instance of this enum for the "info" value.
+     *
+     * @return self
+     */
+    public static function info(): self
+    {
+        return new self(self::SEVERITY_INFO);
+    }
+
+    /**
+     * Creates a new instance of this enum for the "warning" value.
+     *
+     * @return self
+     */
+    public static function warning(): self
+    {
+        return new self(self::SEVERITY_WARNING);
+    }
+
+    /**
+     * Creates a new instance of this enum for the "error" value.
+     *
+     * @return self
+     */
+    public static function error(): self
+    {
+        return new self(self::SEVERITY_ERROR);
+    }
+
+    /**
+     * Creates a new instance of this enum for the "fatal" value.
+     *
+     * @return self
+     */
+    public static function fatal(): self
+    {
+        return new self(self::SEVERITY_FATAL);
+    }
+
+    /**
+     * Returns whether two object instances of this class are equal.
+     *
+     * @param self $other The object to compare
+     *
+     * @return bool
+     */
+    public function isEqualTo(self $other): bool
```

Comment:
This shouldn't be needed, isn't `==` enough?

Reply:
Yes it is, but I think that having an `isEqualTo` method provides a nicer API than comparing two objects to check their internal value is equal

```php
$enum1 = Severity::error();
$enum2 = Severity::fatal();

var_dump($enum1 == $enum2);
var_dump($enum1->isEqualTo($enum2));
```

If someone for some reason uses `===` will always get false from the assertion while using `isEqualTo` is more expressive